src/components/Banner/Banner.tsx

```diff
@@ -1,21 +1,31 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, HTMLProps } from "react";
 import classnames from "classnames";
 
 import "./_banner.scss";
 
-export default function Banner({ isActive = false, children, type }) {
+type Props = {
+  isActive: boolean;
+  children: any;
+  variant: "positive" | "caution" | "negative";
+} & HTMLProps<HTMLElement>;
+
+export default function Banner({
+  isActive,
+  children,
+  variant,
+}: Props): JSX.Element {
   const [bannerClosed, setBannerClosed] = useState(false);
```

Comment:
can you add the type to `bannerClosed` please.

Reply:
This is inferred from the initial state value?

<img width="488" alt="Screenshot 2020-12-01 at 11 07 14" src="https://user-images.githubusercontent.com/505570/100732960-67530500-33c5-11eb-8463-6ff7da28db95.png">

Reply:
![Screenshot 2020-12-01 at 11 24 02](https://user-images.githubusercontent.com/505570/100734562-c6b21480-33c7-11eb-8872-dc6bd0da2605.png)

Reply:
Interesting, must be a special case that it knows what `useState` does.

Reply:
Especially because you can pass a function to `useState` so I wonder if it'll require the type of the return value of the function in that case.